capitalone/rubicon-ml
# make dashboard proxy configurable

closes: #115 
---

## What
* ~exposes `configure_dash_app` which allows users to pass config arguments to `dash.Dash` before the dashboard runs~
* ~changes the dashboard import to `from rubicon_ml.ui.dashboard import Dashboard` to avoid import collisions~
  * ~`configure_dash_app` needs to be run before `Dashboard` is imported as the Dash config is applied at import time~
* rubicon's `Dashboard` class accepts all arguments that can be passed through to `dash.Dash` as `dash_options`
  * thanks @joe-wolfe21 for figuring this out in #117 !
* removes `JupyterDash` in favor of an `IFrame`-based solution for inline viewing in notebooks

## How to Test

```python
from rubicon_ml.ui import Dashboard


dash_options = {"requests_pathname_prefix": "/proxy/8050"}

Dashboard(
    persistence="memory",
    dash_options=dash_options,
).run_server(proxy="http://127.0.0.1:8050::https://my-jupyterhub-server.com/")
```

check out [Dash's API reference](https://dash.plotly.com/reference) for more info on kwargs that can be passed to `dash.Dash` via `configure_dash_app` (like `requests_pathname_prefix`) and to `run_server` (like `proxy`)
